### PR TITLE
Add option to allow task items to be checkable (uncontrolled)

### PR DESCRIPTION
**Pull request by kaspnilsson**
Useful for drafting interactive documents. Could see this expanding to potentially also support passing a callback?

Will happily update documentation if this approach is accepted!

### CODE REVIEW

**Comment by philippkuehn** [comment]
Hey, thanks!

I’m happy to accept a `checkable` callback method to implement custom behavior here. But this should also control if the command is executed to update the `checked` attribute so this could be a fix for https://github.com/ueberdosis/tiptap/issues/2144 too.

An update for the docs would be great too ✌️

**Comment by bkempe** [comment]
I think this could be achieved via:
```ts
        if (!editor.isEditable && onReadOnlyChecked) {
          onReadOnlyChecked(node.attrs.nodeId, checked);
        }
```
after
```ts
if (editor.isEditable && typeof getPos === 'function') { ... }
```

**Comment by kaspnilsson** [comment]
Think I did what y'all requested, let me know if this works @philippkuehn @bkempe!

**Comment by kaspnilsson** [comment]
Ping @philippkuehn @bkempe

**Comment by bkempe** [comment]
@kaspnilsson It's not up to me to approve, but it looks great :)

**Comment by bkempe** [comment]
@kaspnilsson One thing that might be nice: revert the change _if_ `onReadOnlyChecked` returns false?

**Comment by kaspnilsson** [comment]
@bdbch please take a look! :)

**Comment by bdbch** [comment]
Hey @kaspnilsson - looks nice! I think we have some linting issues. Could you resolve them? I'll make sure to take a look again when you're done so we can get your work merged! 👍

**Comment by kaspnilsson** [comment]
@bdbch done!

File: packages/extension-task-item/package.json [hunk 1]
```diff
@@ -23,6 +23,9 @@
   "peerDependencies": {
     "@tiptap/core": "^2.0.0-beta.1"
   },
+  "dependencies": {
+    "prosemirror-model": "^1.16.1"
```
Comment (bdbch) [inline packages/extension-task-item/package.json:27]: Is this dependency needed or should this be a peerDependency?

**Review by bdbch — COMMENTED** [review]
Just a question about the dependencies. Otherwise looks good.

File: packages/extension-task-item/package.json (hunk 1 above)
Comment (kaspnilsson) [reply packages/extension-task-item/package.json:27]: Sure, just needed for the types

Comment (bdbch) [reply packages/extension-task-item/package.json:25]: alright, thanks for the quick answer!

**Pull request merged**